src/os/bluestore/BlueStore.cc

```diff
@@ -1951,17 +1978,61 @@ void BlueStore::ExtentMap::reshard(
       max_blob_end = be;
     }
   }
-  onode->onode.extent_map_shards.swap(new_shard_info);
+  if (new_shard_info.empty() && (si_begin > 0 ||
+				 si_end < shards.size())) {
+    // we resharded a partial range; we must produce at least one output
+    // shard
+    new_shard_info.emplace_back(bluestore_onode_t::shard_info());
```

Comment:
You might want to implement shard_info(offset) ctor and apply it here and throughout?